File: frontend/src/types/certificates.ts
export type CertificateType = 'CONTROL_CLINICO' | 'OBITO' | 'PRESENCIA' | 'CONSENTIMIENTO';

export interface CertificateDraft {
  type: CertificateType;
  patientId: string;
  patientFullName: string;
  patientDni: string;
  hogarId: string;
  hogarName: string;
  hogarAddress: string; // ej: "Palos 295"
  issuedAt: string; // ISO
  bodyText: string;
  doctorDisplayName: string;
  doctorLicenseNumber?: string | null;
}

export interface Certificate {
  id: string;
  resident_id: string;
  facility_id: string;
  certificate_type: CertificateType;
  issued_at: string; // ISO datetime
  issued_by_user_id: string;
  body_text: string;
  content_json?: Record<string, any> | null;
  pdf_url?: string | null;
  created_at: string; // ISO datetime
}

export interface CertificateCreate {
  resident_id: string;
  facility_id: string;
  certificate_type: CertificateType;
  body_text: string;
  issued_at: string; // ISO datetime
  content_json?: Record<string, any> | null;
}

export interface CertificateUpdate {
  body_text?: string;
  issued_at?: string; // ISO datetime
  content_json?: Record<string, any> | null;
}

File: backend/app/schemas/certificates.py
from pydantic import BaseModel
from typing import Dict, Any, Optional
from uuid import UUID
from datetime import datetime


class CertificateCreate(BaseModel):
    resident_id: UUID
    facility_id: UUID
    certificate_type: str  # CONTROL_CLINICO / OBITO / PRESENCIA
    body_text: str  # Texto del cuerpo de la constancia
    issued_at: datetime
    content_json: Optional[Dict[str, Any]] = None  # Campos adicionales opcionales


class CertificateUpdate(BaseModel):
    body_text: Optional[str] = None
    issued_at: Optional[datetime] = None
    content_json: Optional[Dict[str, Any]] = None


class CertificateResponse(BaseModel):
    id: UUID
    resident_id: UUID
    facility_id: UUID
    certificate_type: str
    issued_at: datetime
    issued_by_user_id: UUID
    body_text: str
    content_json: Optional[Dict[str, Any]] = None
    pdf_url: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True
